wdl-engine/src/inputs.rs

```diff
@@ -0,0 +1,756 @@
+//! Implementation of workflow and task inputs.
+
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::BufReader;
+use std::mem;
+use std::path::Path;
+
+use anyhow::Context;
+use anyhow::Result;
+use anyhow::anyhow;
+use anyhow::bail;
+use serde_json::Value as JsonValue;
+use wdl_analysis::document::Document;
+use wdl_analysis::document::Task;
+use wdl_analysis::document::Workflow;
+use wdl_analysis::types::CallKind;
+use wdl_analysis::types::Coercible;
+use wdl_analysis::types::v1::task_hint_types;
+use wdl_analysis::types::v1::task_requirement_types;
+
+use crate::Engine;
+use crate::Value;
+
+/// A type alias to a JSON map (object).
+type JsonMap = serde_json::Map<String, JsonValue>;
+
+/// Represents inputs to a task.
+#[derive(Default, Debug, Clone)]
+pub struct TaskInputs {
+    /// The task input values.
+    inputs: HashMap<String, Value>,
+    /// The overridden requirements section values.
+    requirements: HashMap<String, Value>,
+    /// The overridden hints section values.
+    hints: HashMap<String, Value>,
```

Comment:
I didn't realize you could override hints. What's the use case for that? Spitballing: I guess it makes sense if an implementation leaves _out_ a `hints` section and you want to "fill it in" at runtime. But overriding specified values? That seems wrong to me, but maybe I'm just not thinking of the right use case.

Reply:
Yeah, both requirements and hints [can be overridden](https://github.com/openwdl/wdl/blob/wdl-1.2/SPEC.md#specifying--overriding-requirements-and-hints) and an engine _must_ support doing so.

Probably useful for when you might have a bespoke container image you want to run someone else's task in or the environment you'll be running a task on would benefit from higher minimum/maximum cpu/memory/storage.

Reply:
We've written `workflows` in a way that a user should never have to override a runtime/requirements value (or at least it should be rare), but that use case makes perfect sense to me. A task is failing due to too low a mem allocation? Override `runtime.memory`.

But `hints`??? Those don't seem like something that should be "messed with" by the user. But I haven't actually written any hints sections for production, so maybe I'm just thinking about it wrong.

@adthrasher just curious if you have thoughts on this. Seems an odd case to me.